pages/doc.js

```diff
@@ -109,47 +66,37 @@ export default class Documentation extends Component {
     })
   }
 
-  getLinkHref = (section, file) => {
-    const sectionSlug =
-      sidebar[section].slug || kebabCase(sidebar[section].name)
-    const fileSlug = file ? kebabCase(file.slice(0, -3)) : undefined
-    return `/doc/${compact([sectionSlug, fileSlug]).join('/')}`
+  onNavigate = (path, e) => {
+    e && e.preventDefault()
+    window.history.pushState(null, null, path)
+    this.loadPath(path)
   }
 
-  setCurrentPath = (section, file) => {
-    window.history.pushState(null, null, this.getLinkHref(section, file))
-  }
+  loadStateFromURL = () => this.loadPath(window.location.pathname)
 
-  onSectionSelect = (section, e) => {
-    e && e.preventDefault()
-    const { indexFile, files } = sidebar[section]
-    const file = indexFile || flatten(files)[0]
-    e && this.setCurrentPath(section, indexFile ? undefined : file)
-    this.loadFile({ file, section, parseHeadings: false })
-  }
+  loadPath = path => {
+    const item = getItemByPath(path)
 
-  onFileSelect = (file, section, e) => {
-    e && e.preventDefault()
-    this.setCurrentPath(section, file)
-    this.loadFile({ file, section, parseHeadings: true })
-  }
+    if (!item) {
+      this.setState({ pageNotFound: true, currentItem: {} })
+
+      return
```

Comment:
it's probably better to use `else` instead of an explicit `return`

Reply:
Ok, I will change it.